### ISSUE TITLE
Use several thread to prepare the articles.

### ISSUE THREAD

**Issue by mgautierfr** [openzim/zimwriterfs#14]
For now, we use 3 threads:
- The `directoryVisitor` who visits the source directory and feel a queue (`filenameQueue`)with the filename to add.
- The `main` thread who pops from `filenameQueue`, read the content (open file, read content, parse html, ...), add the content to the zim and create and push a "index token" in a queue for xapianIndexer.
- The `indexer` thread who pop from a "index token" and indexes the content.

From my tests, it seems that the `filenameQueue` is always full, the "index token" queue is always empty
and only one cpu is used (100%). So the `main` thread is always running and the two others are mostly waiting.

The article preparation (open file, read content, parse html, create "index token") is independent from the others. We could run several threads `preparing` articles in parallel.
They will push already prepared articles in new "articles" queue and "index token" for xapianIndexer (as `main` thread is doing)
The `main` thread would only get prepared articles and add it to zim.

It should greatly improve the performance of zimwriterfs.
Except if N is to big (and I/O bound could appears), running N `preparing` threads should globally divide the execution time by N.

**Comment by darkenvy** [comment]
I was going go open a new issue for the request of multi-thread support but found this. I noticed only one thread is ever at 100% and find that it takes 1 hour per GB of finished zim size (10gb zim is 10h for example) on a top of the line i7 overclocked to 4ghz.

Out of curiosity, is it possible to thread more? I'm guessing that due to the lzma2 algorithm we can only have limited threads.
